ludwig/schema/encoders/image_encoders.py

```diff
@@ -284,368 +284,828 @@ class Stacked2DCNNEncoderConfig(BaseEncoderConfig):
-        sorted(list(initializer_registry.keys())),
-        default="xavier_uniform",
-        description="Initializer for the weights matrix.",
-        parameter_metadata=ENCODER_METADATA["ResNetEncoder"]["weights_initializer"],
+    model_variant: Optional[str] = schema_utils.StringOptions(
+        ["base"],
+        default="base",
+        allow_none=False,
+        description="Pretrained model variant to use.",
+        parameter_metadata=ENCODER_METADATA["TVAlexNetEncoder"]["model_variant"],
     )
 
-    output_size: Optional[int] = schema_utils.PositiveInteger(
-        default=128,
-        description="if output_size is not already specified in fc_layers this is the default output_size that will "
-        "be used for each layer. It indicates the size of the output of a fully connected layer. ",
-        parameter_metadata=ENCODER_METADATA["ResNetEncoder"]["output_size"],
+
+@register_encoder_config("convnext_torch", IMAGE)
+@dataclass
+class TVConvNeXtEncoderConfig(TVBaseEncoderConfig):
+    type: str = schema_utils.StringOptions(
+        ["convnext_torch"],
+        default="convnext_torch",
+        allow_none=False,
+        description="Type of encoder.",
     )
 
-    norm: Optional[str] = schema_utils.StringOptions(
-        ["batch", "layer"],
-        default=None,
-        description="if a norm is not already specified in fc_layers this is the default norm that will be used for "
-        "each layer. It indicates the norm of the output and can be null, batch or layer.",
-        parameter_metadata=ENCODER_METADATA["ResNetEncoder"]["norm"],
+    model_variant: Optional[str] = schema_utils.StringOptions(
+        ["tiny", "small", "base", "large"],
+        default="base",
+        allow_none=False,
+        description="Pretrained model variant to use.",
+        parameter_metadata=ENCODER_METADATA["TVConvNeXtEncoder"]["model_variant"],
     )
 
-    norm_params: Optional[Dict[str, Any]] = schema_utils.Dict(
-        default=None,
-        description="parameters used if norm is either batch or layer. For information on parameters used with batch "
-        "see Torch's documentation on batch normalization or for layer see Torch's documentation on layer "
-        "normalization.",
-        parameter_metadata=ENCODER_METADATA["ResNetEncoder"]["norm_params"],
+
+@register_encoder_config("densenet_torch", IMAGE)
+@dataclass
+class TVDenseNetEncoderConfig(TVBaseEncoderConfig):
+    type: str = schema_utils.StringOptions(
+        ["densenet_torch"],
+        default="densenet_torch",
+        allow_none=False,
+        description="Type of encoder.",
     )
 
-    num_fc_layers: Optional[Optional[int]] = schema_utils.PositiveInteger(
-        default=1,
-        description="The number of stacked fully connected layers.",
-        parameter_metadata=ENCODER_METADATA["ResNetEncoder"]["num_fc_layers"],
+    model_variant: Optional[int] = schema_utils.IntegerOptions(
```

Comment:
Want to double-check if this field is really `Optional`? If so, make sure `allow_none` is `True`.

Reply:
removed optional